src/commonMain/kotlin/exchange.dydx.abacus/state/manager/StateManagerAdaptor.kt

```diff
@@ -1318,9 +1332,55 @@ open class StateManagerAdaptor(
             if (success(httpCode) && response != null) {
                 update(stateMachine.historicalFundings(response), oldState)
             }
+            ioImplementations.threading?.async(ThreadingType.main) {
```

Comment:
Can this get called multiple times?  If so, might want to set historicalFundingTimer to null before starting another timer.